README.md

```diff
@@ -1,2 +1,362 @@
+- `__query_create_timestamp` - datetime when the query was scheduled.
+
+#### Job output example
+
+Let's consider the following example.
+
+For query:
+
+```sql
+SELECT
+    `column_a`,
+    `column_b`,
+    `column_c`,
+    `record_time`
+FROM
+    `kafka`.`some-kafka-input-table`
+WHERE
+    `column_d` = 'SOME_VALUE'
+```
+
+the output table may have the following definition:
+
+```sql
+create table `kafka`.`some-flink-table-name` (
+    `column_a` STRING,
+    `column_b` INT,
+    `column_c` STRING,
+    `record_time` TIMESTAMP(3),
+    `__create_timestamp` TIMESTAMP(3),
+    `__query_name` STRING,
+    `__query_description` STRING,
+    `__query_id` STRING,
+    `__query_version` INT,
+    `__query_create_timestamp` TIMESTAMP(3)
+) with (
+    'connector' = 'kafka',
+    'format' = 'json',
+    'topic' = 'some-topic-name',
+    'sink.delivery-guarantee' = 'at-least-once',
+    'properties.bootstrap.servers' = 'example.com:9092'
+)
+;
+```
+
+---
+
+## Deployment
+
+### Infrastructure
+
+Currently, the framework supports only deployment on a Flink Session Cluster running on YARN on AWS EMR cluster.
+
+#### Session Mode
+
+There are two deployment modes in Flink to
+consider: [Application Mode](https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/overview/#application-mode)
+and [Session Mode](https://nightlies.apache.org/flink/flink-docs-master/docs/deployment/overview/#session-mode)
+(Per-Job Mode is deprecated).
+
+- `Application Mode` - a dedicated Flink cluster is created for each Flink application.
+- `Session Mode` - multiple Flink jobs shares the same Flink cluster.
+
+Essentially, Application Mode provides higher resource isolation (separate cluster per job, that is, dedicated
+JobManager, or two if HA is enabled) in comparison to Session Mode, but more resources are utilized. If there are tens
+or hundreds of small jobs running, having a Session Cluster running all the jobs should save a lot of resources.
+
+_Decision_: Flink jobs are deployed on a shared Flink Session Cluster.
+
+### Prerequisites
+
+- Session cluster is up and running. Currently, it is assumed there is only one running Flink session cluster on EMR
+  cluster.
+- All necessary Flink dependencies are available on the classpath on the EMR master node.
+- Python virtualenv with all necessary python dependencies is created on each EMR node.
+  ```bash
+  python3 -m pip install \
+      -r flink-sql-emr-runner/emr-deployment-scripts/jobs-deployment/requirements.txt \
+      -r flink-sql-emr-runner/python/requirements.txt
+  ```
+
+### Running deployment scripts
+
+Sample command to trigger deployment of all queries:
+
+```bash
+python3 flink-sql-emr-runner/emr-deployment-scripts/jobs-deployment/deploy.py \
+    --path /some/path/to/sql/queries/ \
+    --template-file flink-sql-emr-runner/emr-deployment-scripts/job-template.yaml \
+    --pyflink-runner-dir flink-sql-emr-runner/python/ \
+    --external-job-config-bucket some-s3-bucket \
```

Comment:
Add s3 bucket to the Prerequisites section

Reply:
fixed